### PR TITLE
adds get chain with root CA for spiffe ID searching

### PR DESCRIPTION
**Pull request by andrewpmartinez**

### CODE REVIEW

File: ca_pool.go [hunk 1]
```diff
@@ -35,6 +35,7 @@ func (self *CaPool) isSelfSignedCA(cert *x509.Certificate) bool {
 	return cert.IsCA && cert.CheckSignatureFrom(cert) == nil
 }
 
+// GetChainMinusRoot returns a chain from `cert` up to, but not including, the root CA if possible
```
Comment (dovholuknf) [inline ca_pool.go:38]: what if it's "not possible"?

File: ca_pool.go [hunk 2]
```diff
@@ -53,6 +54,30 @@ func (self *CaPool) GetChainMinusRoot(cert *x509.Certificate, extraCerts ...*x50
 	}
 }
 
+// GetChain returns a chain from `cert` up and including the root CA if possible
+func (self *CaPool) GetChain(cert *x509.Certificate, extraCerts ...*x509.Certificate) []*x509.Certificate {
+	var result []*x509.Certificate
+	result = append(result, cert)
```
Comment (dovholuknf) [inline ca_pool.go:60]: should this nil check?

**Review by dovholuknf — APPROVED** [review]

File: ca_pool.go (hunk 1 above)
Comment (andrewpmartinez) [reply ca_pool.go]: documented

File: ca_pool.go (hunk 2 above)
Comment (andrewpmartinez) [reply ca_pool.go:70]: nil is now checked

**Pull request merged**